src/commands/importDocuments.ts

```diff
@@ -0,0 +1,134 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { NewDocument } from 'documentdb';
+import * as fse from 'fs-extra';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, IAzureParentNode, parseError } from 'vscode-azureextensionui';
+import { DocDBCollectionTreeItem } from '../docdb/tree/DocDBCollectionTreeItem';
+import { DocDBDocumentsTreeItem } from '../docdb/tree/DocDBDocumentsTreeItem';
+import { ext } from '../extensionVariables';
+import { MongoCollectionTreeItem } from '../mongo/tree/MongoCollectionTreeItem';
+
+export async function importDocuments(tree: AzureTreeDataProvider, uris: vscode.Uri[] | undefined, collectionNode: IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem> | undefined): Promise<void> {
+    if (!uris) {
+        uris = await askForDocuments();
+    }
+    let ignoredUris: vscode.Uri[] = []; //account for https://github.com/Microsoft/vscode/issues/59782
+    uris = uris.filter((uri) => {
+        if (uri.fsPath.endsWith('.json')) {
+            return true;
+        } else {
+            ignoredUris.push(uri);
+            return false;
+        }
+    });
+    if (ignoredUris.length) {
+        ext.outputChannel.appendLine(`The following selected files are not json: ${ignoredUris.map(uri => uri.fsPath).join(',')}. \nIgnoring these.`);
+        ext.outputChannel.show();
+    }
+    if (!collectionNode) {
+        collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+    }
+
+    const documents = await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: "Import documents..."
+        },
+        (_progress) => parseDocumentsForErrors(uris)
+    );
+
+    let result: string;
+    if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+        const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+        result = await collectionTreeItem.executeCommand('insertMany', [JSON.stringify(documents)]);
+    } else {
+        result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, uris);
+    }
+    await collectionNode.refresh();
+    await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+    return await ext.ui.showOpenDialog({
+        canSelectMany: true,
+        openLabel: "Import",
+        filters: {
+            "JSON": ["json"]
+        },
+        defaultUri: vscode.Uri.file(vscode.workspace.rootPath)
+    });
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+    const parseResult = await parseDocuments(nodes);
```

Comment:
nit: These three lines can just be `const [documents, hasErrors] = await parseDocuments(nodes);`

Reply:
I remember python allows this, but didn't realize TS would be fine with this too. Thanks for the headsup!